backend/infrahub/proposed_change/tasks.py

```diff
@@ -2,15 +2,109 @@
 
 from infrahub_sdk.protocols import CoreProposedChange
 from prefect import flow, task
+from prefect.client.schemas.objects import (
+    State,  # noqa: TCH002
+)
 from prefect.logging import get_run_logger
+from prefect.states import Completed, Failed
 
-from infrahub.core.constants import ProposedChangeState
+from infrahub.core import registry
+from infrahub.core.branch import Branch
+from infrahub.core.branch.tasks import merge_branch
+from infrahub.core.constants import InfrahubKind, ValidatorConclusion
+from infrahub.core.protocols import CoreDataCheck, CoreValidator
+from infrahub.core.protocols import CoreProposedChange as InternalCoreProposedChange
+from infrahub.proposed_change.constants import ProposedChangeState
 from infrahub.services import (
     services,
 )
 
 
-@flow(name="proposed-changes-cancel-branch", description="Cancel all Proposed change associated with a branch.")
+async def _proposed_change_transition_state(
+    state: ProposedChangeState,
+    proposed_change: InternalCoreProposedChange | None = None,
+    proposed_change_id: str | None = None,
+) -> None:
+    service = services.service
+    async with service.database.start_session() as db:
+        if proposed_change is None and proposed_change_id:
+            proposed_change = await registry.manager.get_one(
+                db=db, id=proposed_change_id, kind=InternalCoreProposedChange, raise_on_error=True
+            )
+        if proposed_change:
+            proposed_change.state.value = state.value  # type: ignore[misc]
+            await proposed_change.save(db=db)
+
+
+# async def proposed_change_transition_merged(flow: Flow, flow_run: FlowRun, state: State) -> None:
+#     await _proposed_change_transition_state(
+#         proposed_change_id=flow_run.parameters["proposed_change_id"], state=ProposedChangeState.MERGED
+#     )
+
+
+# async def proposed_change_transition_open(flow: Flow, flow_run: FlowRun, state: State) -> None:
+#     await _proposed_change_transition_state(
+#         proposed_change_id=flow_run.parameters["proposed_change_id"], state=ProposedChangeState.OPEN
+#     )
+
+
+@flow(
+    name="proposed-change-merge",
+    flow_run_name="Merge propose change: {proposed_change_name} ",
+    description="Merge a given proposed change.",
+    # TODO need to investigate why these function are not working as expected
```

Comment:
Not sure why the callback are not working, I'll revisit that after upgrading prefect to the latest version to exclude potential bugs